modules/light-clients/07-tendermint/migrations.go

```diff
@@ -0,0 +1,72 @@
+package tendermint
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/cosmos/cosmos-sdk/codec"
+	"github.com/cosmos/cosmos-sdk/store/prefix"
+	storetypes "github.com/cosmos/cosmos-sdk/store/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+
+	clienttypes "github.com/cosmos/ibc-go/v6/modules/core/02-client/types"
+	host "github.com/cosmos/ibc-go/v6/modules/core/24-host"
+	"github.com/cosmos/ibc-go/v6/modules/core/exported"
+)
+
+// PruneTendermintConsensusStates prunes all expired tendermint consensus states. This function
+// may optionally be called during in-place store migrations. The ibc store key must be provided.
+func PruneTendermintConsensusStates(ctx sdk.Context, cdc codec.BinaryCodec, storeKey storetypes.StoreKey) error {
+	store := ctx.KVStore(storeKey)
+
+	// iterate over ibc store with prefix: clients/07-tendermint,
+	tendermintClientPrefix := []byte(fmt.Sprintf("%s/%s", host.KeyClientStorePrefix, exported.Tendermint))
+	iterator := sdk.KVStorePrefixIterator(store, tendermintClientPrefix)
+
+	var clients []string
+
+	// collect all clients to avoid performing store state changes during iteration
+	defer iterator.Close()
+	for ; iterator.Valid(); iterator.Next() {
+		path := string(iterator.Key())
+		if !strings.Contains(path, host.KeyClientState) {
+			// skip non client state keys
+			continue
+		}
+
+		clientID := host.MustParseClientStatePath(path)
+		clients = append(clients, clientID)
+	}
+
+	// keep track of the total consensus states pruned so chains can
+	// understand how much space is saved when the migration is run
+	var totalPruned int
+
+	for _, clientID := range clients {
+		clientPrefix := []byte(fmt.Sprintf("%s/%s/", host.KeyClientStorePrefix, clientID))
```

Comment:
qq: why does this prefix have a trailing slash while the tendermintClientPrefix above does not?

Reply:
example full key:

`clients/07-tendermint-0/clientState`

tendermintClientPrefix: `clients/07-tendermint`
clientStore: `clients/07-tendermint-0/`
clientState key: `clientState`

Does that make sense? The tendermint client prefix is iterating on the tendermint client type, The next value in the key is `-{N}`